pacupg/pacupg

```diff
@@ -79,38 +79,41 @@ aur(){
         warn "pacaur not found. Aborting\n"
         exit 1
     fi
-    source $pacaurconfig
-    [[ -z $clean || $clean != "true" ]] &&  
-        sudo sed -i '/clean=/d' $pacaurconfig && 
-        echo "clean=true" | sudo tee -a $pacaurconfig > /dev/null # Make sure clean option is off in pacaur
-
-    builddir=${builddir:-/tmp/pacaurtmp-$USER}
+    source "${pacaurconfig}"
+    [[ -z "${clean}" || "${clean}" != "true" ]] &&
+        sudo sed -i '/clean=/d' "${pacaurconfig}" &&
+        echo "clean=true" | sudo tee -a "${pacaurconfig}" > /dev/null # Make sure clean option is off in pacaur
 
+    builddir="${builddir:-${TMPDIR:-/tmp}/pacaurtmp-$USER}"
 
     msg "Checking for updates...\n"
-    aurpkgs=$(pacaur --color never $ignoreopts $devel --check | awk -F'aur  ' '{print $2}')
-    if [[ -z $aurpkgs ]]; then
+    aurpkgs="$(pacaur --color never ${ignoreopts} ${devel} --check | awk -F'aur  ' '{print $2}')"
+    if [[ -z "${aurpkgs}" ]]; then
         msg "No AUR packages to upgrade.\n"
-        [[ $pkgsupgraded == 1 ]] && { btrgrub;  msg "Log of upgraded packages available at ${log_file}\n"; return; }
+        [[ "${pkgsupgraded}" == 1 ]] && { btrgrub;  msg "Log of upgraded packages available at ${log_file}\n"; return; }
         return 2
     fi
-    rm -rf "$builddir" && mkdir "$builddir"
-    if eval "pacaur --color always -Syuw $devel"; then
-    buildpkgs=$(find $builddir -type f -regex ".*\.pkg\.\(tar\|tar\.[gx]z\)")
-    if [[ -n $buildpkgs ]]; then
+    rm -rf "${builddir}" && mkdir "${builddir}"
+
+    # Make sure the packages land in the folder we expect by setting PKGDEST before building the packages.
+    # PKGDEST is used by makepkg as a target location for built packages (see man makepkg.conf).
+    export PKGDEST="${builddir}"
+    if eval "pacaur --color always -Syuw ${devel}"; then
+    buildpkgs="$(find ${builddir} -type f -regex ".*\.pkg\.\(tar\|tar\.[gx]z\)")"
+    if [[ "${buildpkgs}" ]]; then
             eval "in_block=1"
             # Take snapshot and store snapshot number in $pre
-            pre=$(sudo snapper create --type=pre --cleanup-algorithm=number --print-number --description="AUR upgrade") 
+            pre="$(sudo snapper create --type=pre --cleanup-algorithm=number --print-number --description="AUR upgrade")"
```

Comment:
Generally good practice, yes, but not needed since the "--print-number" option only outputs a single digit with no spaces.